profiling/src/internal/upscaling.rs

```diff
@@ -85,7 +86,7 @@ impl UpscalingRules {
                 let (_, rules) = self
                     .rules
                     .get_index_mut(index)
-                    .expect("Already existing rules");
+                    .context("Already existing rules")?;
```

Comment:
Maybe we can improve the message at the same time? Something like:

```suggestion
                    .context("Expected upscaling rule to exist for index {index}")?;
```